plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,408 @@
+
+- name: Query a MACSec Policy with mac_sec_policy uuid
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy: ansible_test_mac_sec_policy
+    state: query
+  register: query_uuid
+
+- name: Delete a MACSec Policy
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy: ansible_test_mac_sec_policy
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import NDO_CIPHER_SUITE_MAP, NDO_SECURITY_POLICY_MAP
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            mac_sec_policy=dict(type="str", aliases=["name"]),
+            mac_sec_policy_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            type=dict(type="str", choices=["fabric", "access"], default="fabric"),
+            cipher_suite=dict(type="str", choices=["128_gcm_aes", "128_gcm_aes_xpn", "256_gcm_aes", "256_gcm_aes_xpn"]),
+            window_size=dict(type="int"),
+            security_policy=dict(type="str", choices=["should_secure", "must_secure"]),
+            sak_expiry_time=dict(type="int"),
+            confidentiality_offset=dict(type="int"),
+            key_server_priority=dict(type="int"),
+            mac_sec_key=dict(
+                type="list",
+                elements="dict",
+                options=dict(
+                    key_name=dict(type="str", required=True),
+                    psk=dict(type="str", required=True, no_log=True),
+                    start_time=dict(type="str"),
+                    end_time=dict(type="str"),
+                ),
+                no_log=False,
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["mac_sec_policy"]],
+            ["state", "absent", ["mac_sec_policy"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    mac_sec_policy = module.params.get("mac_sec_policy")
+    mac_sec_policy_uuid = module.params.get("mac_sec_policy_uuid")
+    description = module.params.get("description")
+    admin_state = module.params.get("admin_state")
+    type = module.params.get("type")
```

Comment:
should we use type as it is a built-in function

Reply:
renamed type to interface_type